burnman/layer.py

```diff
@@ -0,0 +1,797 @@
+from __future__ import print_function
+# This file is part of BurnMan - a thermoelastic and thermodynamic toolkit for the Earth and Planetary Sciences
+# Copyright (C) 2012 - 2017 by the BurnMan team, released under the GNU
+# GPL v2 or later.
+
+import numpy as np
+from scipy.integrate import odeint
+from scipy.integrate import quad
+from scipy.interpolate import UnivariateSpline
+from burnman import averaging_schemes
+from burnman import constants
+from burnman import geotherm
+import warnings
+
+
+from .material import Material, material_property
+from .mineral import Mineral
+from .composite import Composite
+from .seismic import Seismic1DModel
+
+import matplotlib.pyplot as plt
+
+
+class Layer(object):
+    """
+    A planetary layer class
+    """
+
+    def __init__(self, name=None, radius_planet=None,
+                 min_depth=None, max_depth=None, n_slices=None, verbose=False):
+
+        self.name = name
+        self.min_depth = min_depth
+        self.max_depth = max_depth
+        self.thickness = self.max_depth - self.min_depth
+        self.n_slices = n_slices
+        self.depths = np.linspace(min_depth, max_depth, n_slices)
```

Comment:
On balance, I think I'd prefer to see a single "depth slices" in the initialiser which takes an array of slice depths. That way, the user can provide non-uniform slices, and this also meshes more nicely with the user_defined temperature/pressure inputs below.